### PR TITLE
Add kubernetes-filebeat

### PR DESCRIPTION
**Pull request by tauffredou**

### CODE REVIEW

File: kubernetes-filebeat/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+appVersion: "1.0"
+description: A Helm chart for Kubernetes
```
Comment (gearnode) [inline kubernetes-filebeat/Chart.yaml]: Use the real description here.

File: kubernetes-filebeat/templates/configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "kubernetes-filebeat.fullname" . }}-config
+  labels:
+    {{- if .Values.cluster_service }}
+    k8s-app: {{ template "kubernetes-filebeat.fullname" . }}
```
Comment (gearnode) [inline kubernetes-filebeat/templates/configmap.yaml]: Why do you introduce a new label ?

File: kubernetes-filebeat/templates/configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        # Reload prospectors configs as they change:
+        reload.enabled: false
+      modules:
+        path: ${path.config}/modules.d/*.yml
+        # Reload module configs as they change:
+        reload.enabled: false
+    processors:
+      - add_cloud_metadata:
+
+    {{- if .Values.app.elastic_cloud_enabled }}
```
Comment (gearnode) [inline kubernetes-filebeat/templates/configmap.yaml]: We don't use elastic-cloud and we can't test this configuration. I think we can drop this part of this configuration to improve the readability.

File: kubernetes-filebeat/templates/configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+data:
+  kubernetes.yml: |-
+    - type: docker
+      containers.ids:
+      - "*"
+      processors:
+        - add_kubernetes_metadata:
+            in_cluster: true
```
Comment (gearnode) [inline kubernetes-filebeat/templates/configmap.yaml]: Style

File: kubernetes-filebeat/templates/daemonset.yaml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+          name: {{ template "kubernetes-filebeat.fullname" . }}-config
+      - name: varlibdockercontainers
+        hostPath:
+          path: /var/lib/docker/containers
+      - name: prospectors
+        configMap:
+          defaultMode: 0600
+          name: {{ template "kubernetes-filebeat.fullname" . }}-prospectors
+      - name: data
+        emptyDir: {}
```
Comment (gearnode) [inline kubernetes-filebeat/templates/daemonset.yaml]: Style

Comment (tauffredou) [reply kubernetes-filebeat/templates/daemonset.yaml]: ?

File: kubernetes-filebeat/templates/configmap.yaml (hunk 2 above)
Comment (tauffredou) [reply kubernetes-filebeat/templates/configmap.yaml]: k8s-app is a special one; it allows labeled apps to appear in the "cluster-info"

**Pull request merged**